nodejs/awsx/cloudwatch/dashboard.ts

```diff
@@ -88,20 +94,20 @@ export class Dashboard extends aws.cloudwatch.Dashboard {
      * be treated as a single row to add to the grid.
      */
     constructor(name: string, args: DashboardArgs, opts: pulumi.CustomResourceOptions = {}) {
+        const region = utils.ifUndefined(args.region, utils.getRegionFromOpts(opts));
+
         super(name, {
             dashboardName: utils.ifUndefined(args.name, name),
-            dashboardBody: getDashboardBody(args, opts).apply(b => JSON.stringify(b)),
+            dashboardBody: getDashboardBody(args, region).apply(b => JSON.stringify(b)),
         }, opts);
 
-        const region = utils.getRegion(this);
-
         this.url = pulumi.interpolate
             `https://${region}.console.aws.amazon.com/cloudwatch/home?region=${region}#dashboards:name=${this.dashboardName}`;
     }
 }
 
 /** @internal */
-export function getDashboardBody(args: DashboardArgs, opts: pulumi.CustomResourceOptions = {}) {
+export function getDashboardBody(args: DashboardArgs, region: pulumi.Output<aws.Region>) {

```

Comment:
this is just the helper that hte main Dashboard code calls.  It's not exposed to users (hence @internal).  We just have it exported so tests can call it.